src/mapping/RadialGeoMultiscaleMapping.cpp

```diff
@@ -0,0 +1,217 @@
+    if (_type == MultiscaleType::SPREAD) {
+      // Determine principle axis midpoints as borders to assign 3D vertices to their respective 1D vertex
+      Eigen::VectorXd axisMidpoints(inSize);
+      auto &          inputVerticesRef = input()->vertices();
+      // Order the vertices of the 1D input mesh, to correctly compute the midpoints of the segments between neighboring vertices
+      std::vector<size_t> ordered_vertex_indices(input()->vertices().size());
+      std::iota(ordered_vertex_indices.begin(), ordered_vertex_indices.end(), 0);
+      std::sort(ordered_vertex_indices.begin(), ordered_vertex_indices.end(),
+                [effectiveCoordinate, &inputVerticesRef](const size_t aindex, const size_t bindex) {
+                  return inputVerticesRef[aindex].getCoords()[effectiveCoordinate] < inputVerticesRef[bindex].getCoords()[effectiveCoordinate];
+                });
+      // Compute the midpoints of the 1D input mesh
+      for (size_t i = 0; i < (inSize - 1); i++) {
+        auto axisPositionCurrent = inputVerticesRef[ordered_vertex_indices[i]].getCoords()[effectiveCoordinate];
+        auto axisPositionNext    = inputVerticesRef[ordered_vertex_indices[i] + 1].getCoords()[effectiveCoordinate];
+        axisMidpoints(i)         = (axisPositionCurrent + axisPositionNext) / 2;
+      }
+      axisMidpoints(inSize - 1) = std::numeric_limits<double>::max(); // large number, such that vertices after the last midpoint are still assigned
+
+      /*
+        3D vertices are projected onto the 1D axis and the data is then mapped
+        to the nearest neighbors of the 1D vertices in projection space.
+      */
+      _vertexIndicesSpread.clear();
+      _vertexIndicesSpread.reserve(output()->vertices().size());
+      for (size_t i = 0; i < outSize; i++) {
+        auto vertexCoord = output()->vertices()[i].getCoords()[effectiveCoordinate];
+        int  index       = 0;
+        while (vertexCoord > axisMidpoints(index)) {
+          PRECICE_ASSERT(index + 1 < static_cast<int>(inSize));
+          index++;
+        }
+        _vertexIndicesSpread.push_back(index);
+      }
+    } else {
+      PRECICE_ASSERT(_type == MultiscaleType::COLLECT)
+      /*
+        3D vertices are projected onto the 1D axis and the data is then mapped
+        to (and averaged at) the nearest 1D vertex in projection space.
+      */
+
+      // determine principle axis midpoints as borders to assign 3D vertices to their respective 1D vertex
+      Eigen::VectorXd axisMidpoints(outSize);
+      auto &          outputVerticesRef = output()->vertices();
+
+      // Order the vertices of the 1D output mesh, to correctly compute the midpoints of the segments between neighboring vertices
+      std::vector<size_t> ordered_vertex_indices(output()->vertices().size());
+      std::iota(ordered_vertex_indices.begin(), ordered_vertex_indices.end(), 0);
+      std::sort(ordered_vertex_indices.begin(), ordered_vertex_indices.end(),
```

Comment:
Do we need to preserve order of elements with the same effectiveCoordinate? If so, then we need `std::stable_sort` here.